### PR TITLE
Created initial graph component

### PR DESCRIPTION
**Pull request by yoavf11**
PLEASE NOTE
There is a graph example in 'App.jsx'

### CODE REVIEW

File: src/components/Graph/Graph.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    <div
+      style={
+        {
+          position: 'relative',
+          width: '500px',
+          height: '500px',
+          border: '1px solid black'
+        }
+      }>
+      {renderNodes}
```
Comment (Tamir198) [inline src/components/Graph/Graph.jsx]: Avoid inline styles, use CSS (I recommend css modules over global css)

File: src/components/Graph/Graph.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+/*
+SVG OPTION
+
+import React, { useRef, useEffect } from 'react';
+
+const Graph = ({ nodes, edges }) => {
+  const canvasRef = useRef(null);
+
+  useEffect(() => {
```
Comment (Tamir198) [inline src/components/Graph/Graph.jsx]: Remove all those comments

File: src/components/Graph/Node.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+const Node = ({ x, y }) => {
+  return (
+    <div
+      style={{
+        position: 'absolute',
+        width: '20px',
```
Comment (Tamir198) [inline src/components/Graph/Node.jsx:8]: Also avoid inline styles in here as well

**Review by Tamir198 — CHANGES_REQUESTED** [review]
Good work, Please remove the graph example from the `App.jsx`. 
Also Can you add image of the graph to this pr for the rest of the team? 

Instead of using `nodes.map((node, index)` you can create custom generic list component and use it instead or repeating the same code.

File: src/components/Graph/Graph.jsx (hunk 1 above)
Comment (yoavf11) [reply src/components/Graph/Graph.jsx]: That it temporary until we figure out which library we use

**Comment by yoavf11** [comment]
> Good work, Please remove the graph example from the `App.jsx`. Also Can you add image of the graph to this pr for the rest of the team?
> 
> Instead of using `nodes.map((node, index)` you can create custom generic list component and use it instead or repeating the same code.

Screenshot 2024-04-29 012226 That's an example graph.

Will fix the  `nodes.map((node, index)` soon

File: src/components/CreateGraph/CreateGraph.jsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import Graph from '../Graph/Graph';
+
+const CreateGraph = () => {
+    const [nodes, setNodes] = useState([]);
+    const [edges, setEdges] = useState([]);
+
+    useEffect(() => {
+        const storedNodes = JSON.parse(localStorage.getItem('graphNodes')) || [];
+        const storedEdges = JSON.parse(localStorage.getItem('graphEdges')) || [];
+
```
Comment (Tamir198) [inline src/components/CreateGraph/CreateGraph.jsx:13]: Extract the local storage usage into a service please

File: src/components/CreateGraph/CreateGraph.jsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        if (storedNodes.length > 0 && storedEdges.length > 0) {
+            setNodes(storedNodes);
+            setEdges(storedEdges);
+        }
+    }, []);
+
+    useEffect(() => {
+        localStorage.setItem('graphNodes', JSON.stringify(nodes));
+        localStorage.setItem('graphEdges', JSON.stringify(edges));
+    }, [nodes, edges]);
```
Comment (Tamir198) [inline src/components/CreateGraph/CreateGraph.jsx:23]: Same comment about local storage service in here

File: src/components/CreateGraph/CreateGraph.jsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+            message.error('Node already exists.');
+            return;
+        }
+        setNodes([...nodes, newNode]);
+    };
+
+    const addEdge = () => {
+        if (nodes.length < 2) {
+            message.error("Can't add edge. There must be at least two nodes.");
+            return;
```
Comment (Tamir198) [inline src/components/CreateGraph/CreateGraph.jsx:38]: Please extract this into constant file (check with noa about the structure, she is already working on constant folder)

File: src/components/CreateGraph/CreateGraph.jsx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+        }
+        const randomSource = Math.floor(Math.random() * nodes.length);
+        let randomTarget = Math.floor(Math.random() * nodes.length);
+        while (randomTarget === randomSource) {
+            randomTarget = Math.floor(Math.random() * nodes.length);
+        }
```
Comment (Tamir198) [inline src/components/CreateGraph/CreateGraph.jsx:44]: Extract this `Math.floor(Math.random() * nodes.length)` into a function so you wont repeat yourself

File: src/components/CreateGraph/CreateGraph.jsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        const newEdge = { source: nodes[randomSource], target: nodes[randomTarget] };
+        const edgeExists = edges.some(edge => (
+            (edge.source === newEdge.source && edge.target === newEdge.target) ||
+            (edge.source === newEdge.target && edge.target === newEdge.source)
+        ));
+        if (edgeExists) {
+            message.error('Edge already exists.');
+            return;
```
Comment (Tamir198) [inline src/components/CreateGraph/CreateGraph.jsx:52]: Same comment about the constant file

File: src/components/CreateGraph/CreateGraph.jsx [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    };
+
+    return (
+        <>
+            <Flex gap="L" align="center" vertical>
+                <Flex gap="small" wrap="wrap">
+                    <Button type="primary" icon={<PlusCircleFilled />} size={'large'} onClick={addNode}>
+                        Add Node
+                    </Button>
+                    <Button type="primary" icon={<PlusCircleFilled />} size={'large'} onClick={addEdge}>
```
Comment (Tamir198) [inline src/components/CreateGraph/CreateGraph.jsx:89]: You can extract this into component so you wont repeat yourself

File: src/components/Graph/Graph.jsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import Node from './Node';
+import Edge from './Edge';
+
+const ListItem = ({ items, renderItem }) => {
+  return items.map((item, index) => renderItem(item, index));
+};
```
Comment (Tamir198) [inline src/components/Graph/Graph.jsx:7]: You call this ListItem but this is the entire list. 
If you want you can create a generic list it should expect a component to render and data to map over

**Review by Tamir198 — CHANGES_REQUESTED** [review]
Generally, you have a lot of hardcoded texts - extract them to constant file (talk to noa about the structure)

**Pull request closed without merging**